crates/bevy_core_pipeline/src/oit/resolve/mod.rs

```diff
@@ -0,0 +1,211 @@
+            Shader::from_wgsl
+        );
+
+        let Some(render_app) = app.get_sub_app_mut(RenderApp) else {
+            return;
+        };
+
+        render_app.add_systems(
+            Render,
+            (
+                queue_oit_resolve_pipeline.in_set(RenderSet::Queue),
+                prepare_oit_resolve_bind_group.in_set(RenderSet::PrepareBindGroups),
+            ),
+        );
+    }
+
+    fn finish(&self, app: &mut bevy_app::App) {
+        let Some(render_app) = app.get_sub_app_mut(RenderApp) else {
+            return;
+        };
+
+        render_app.init_resource::<OitResolvePipeline>();
+    }
+}
+
+/// Bind group for the OIT resolve pass
+#[derive(Resource, Deref)]
+pub struct OitResolveBindGroup(pub BindGroup);
+
+/// Bind group layouts used for the OIT resolve pass
+#[derive(Resource)]
+pub struct OitResolvePipeline {
+    /// View bind group layout
+    pub view_bind_group_layout: BindGroupLayout,
+    /// Depth bind group layout
+    pub oit_depth_bind_group_layout: BindGroupLayout,
+}
+
+impl FromWorld for OitResolvePipeline {
+    fn from_world(world: &mut World) -> Self {
+        let render_device = world.resource::<RenderDevice>();
+
+        let view_bind_group_layout = render_device.create_bind_group_layout(
+            "oit_resolve_bind_group_layout",
+            &BindGroupLayoutEntries::sequential(
+                ShaderStages::FRAGMENT,
+                (
+                    uniform_buffer::<ViewUniform>(true),
+                    // layers
+                    storage_buffer_sized(false, None),
+                    // layer ids
+                    storage_buffer_sized(false, None),
+                ),
+            ),
+        );
+
+        let oit_depth_bind_group_layout = render_device.create_bind_group_layout(
+            "oit_depth_bind_group_layout",
+            &BindGroupLayoutEntries::single(ShaderStages::FRAGMENT, texture_depth_2d()),
+        );
+        OitResolvePipeline {
+            view_bind_group_layout,
+            oit_depth_bind_group_layout,
+        }
+    }
+}
+
+#[derive(Component, Deref, Clone, Copy)]
+pub struct OitResolvePipelineId(pub CachedRenderPipelineId);
+
+/// This key is used to cache the pipeline id and to specialize the render pipeline descriptor
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub struct OitResolvePipelineKey {
+    hdr: bool,
+}
+
+#[allow(clippy::too_many_arguments)]
+pub fn queue_oit_resolve_pipeline(
+    mut commands: Commands,
+    pipeline_cache: Res<PipelineCache>,
+    resolve_pipeline: Res<OitResolvePipeline>,
+    views: Query<(Entity, &ExtractedView), With<OrderIndependentTransparencySettings>>,
+    // Store the key with the id to make the clean up logic easier
+    // This also means it will always replace the entry if the key changes so nothing to clean up
+    mut cached_pipeline_id: Local<HashMap<Entity, (OitResolvePipelineKey, CachedRenderPipelineId)>>,
+) {
+    let mut current_view_entities = vec![];
```

Comment:
Probs not a big deal with so few views but this should probably be `EntityHashSet`